Trustroots/trustroots
# Refactor Navigation component

#### Proposed Changes

- move `Navigation` component from `references` to `core`
- add navigation for small screens
- block buttons based on provided (validation) errors
- add tooltip with error message to blocked buttons
- use in `CreateReference` component

#### Testing

- `git switch refactor-navigation`
- make sure you have references enabled: `{ featureFlags: { reference: true } }` in `/config/env/development.js`
- go to `/profile/{some-username}/references/new`
- see that navigation works on a big screen
  ![image](https://user-images.githubusercontent.com/7449720/78503061-9ba5f100-7764-11ea-8814-05be11b025b1.png)
- see that navigation works on a small screen
  ![image](https://user-images.githubusercontent.com/7449720/78503120-e58ed700-7764-11ea-8f38-73be7b793dbc.png)
- hover disabled buttons and see validation errors in tooltip
- see whether `modules/core/client/components/README.md` makes sense

~Depends on #1364, which should be merged first~
Preparation for #1350
